### ISSUE TITLE
Performance degradation on GatherElements

### ISSUE THREAD

**Issue by xiaojul** [microsoft/onnxruntime#11195]
**Describe the bug**
The GatherElements node increases the latency a lot. For a simple onnx model only has GatherElements node, the latency is >10x comparing to the pytorch model.
pytorch model: 0.007374s after 100 iterations
onnx model: 0.152331s after 100 iterations

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 20.04
- ONNX Runtime installed from (source or binary): pip3
- ONNX Runtime version: 1.11.0
- Python version: 3.9.1

**To Reproduce**
- Describe steps/code to reproduce the behavior.
```
import onnxruntime
import torch
import torch.nn as nn
import time

device = torch.device("cpu")

class DummyNet(nn.Module):
    def forward(self, x, y):
        z = torch.gather(x, 1, y)
        return z

dummy_model = DummyNet()
dummy_model.to(device)
dummy_model.eval()

x = torch.rand(1,4,353,128, dtype=torch.float)
y = torch.randint(0, 1, (1,4,353,128), dtype=torch.long)

with torch.no_grad():
    torch.onnx.export(dummy_model,(x,y),"./test.onnx",
                      do_constant_folding=True, verbose=False, opset_version=11,
                      input_names=["x", "y"],output_names=["z"])

count = 100

dummy_model.eval()

## test pytorch model latency
with torch.no_grad():
    start = time.perf_counter()
    for i in range(count):
        _ = dummy_model(x, y)
    end = time.perf_counter() - start
print('pytorch model: {:.6f}s after 100 iterations'.format(end))

## test onnx latency
test_session = onnxruntime.InferenceSession("./test.onnx")
inputs = {
    "x": x.detach().cpu().numpy() if x.requires_grad else x.cpu().numpy(),
    "y": y.detach().cpu().numpy() if y.requires_grad else y.cpu().numpy()
}
with torch.no_grad():
    start = time.perf_counter()
    for i in range(count):
        _ = test_session.run(None, inputs)
    end = time.perf_counter() - start
print('onnx model: {:.6f}s after 100 iterations'.format(end))
```

**Expected behavior**
I would expect a comparable behavior

**Screenshots**
<img width="265" alt="Capture" src="https://user-images.githubusercontent.com/64522260/163070404-f3f0ae47-c1b3-4ff3-9912-73984583c1a6.PNG">

**Comment by yufenglee** [comment]
The GatherElement implementation is not efficient. It is copying one element one time. Actually, we can copy a chunk of elements once. For example, for the attached model, it has shape [1x4x353x128] and gathering axis is on 1. We can copy 353x128 at once and it needs only 4 memory chunk copy.

For a general case with shape [D0, D1, …, Dk,…,Dn-1] and axis k, we can copy chunk Dk+1 * Dk+2 *….* Dn-1 at once.

**Comment by yufenglee** [comment]
> The GatherElement implementation is not efficient. It is copying one element one time. Actually, we can copy a chunk of elements once. For example, for the attached model, it has shape [1x4x353x128] and gathering axis is on 1. We can copy 353x128 at once and it needs only 4 memory chunk copy.
> 
> For a general case with shape [D0, D1, …, Dk,…,Dn-1] and axis k, we can copy chunk Dk+1 * Dk+2 _…._ Dn-1 at once.

the claim is incorrect. We still need copy element by element, but assignment is faster than memcpy.
